aiida_common_workflows/workflows/relax/gpaw/generator.py

```diff
@@ -0,0 +1,150 @@
+        electronic_type: Union[ElectronicType, str] = ElectronicType.METAL,
+        spin_type: Union[SpinType, str] = SpinType.NONE,
+        magnetization_per_site: Union[List[float], Tuple[float]] = None,
+        threshold_forces: float = None,
+        threshold_stress: float = None,
+        reference_workchain=None,
+        **kwargs
+    ) -> engine.ProcessBuilder:
+        """Return a process builder for the corresponding workchain class with inputs set according to the protocol.
+
+        :param structure: the structure to be relaxed.
+        :param engines: a dictionary containing the computational resources for the relaxation.
+        :param protocol: the protocol to use when determining the workchain inputs.
+        :param relax_type: the type of relaxation to perform.
+        :param electronic_type: the electronic character that is to be used for the structure.
+        :param spin_type: the spin polarization type to use for the calculation.
+        :param magnetization_per_site: a list with the initial spin polarization for each site. Float or integer in
+            units of electrons. If not defined, the builder will automatically define the initial magnetization if and
+            only if `spin_type != SpinType.NONE`.
+        :param threshold_forces: target threshold for the forces in eV/Å.
+        :param threshold_stress: target threshold for the stress in eV/Å^3.
+        :param reference_workchain: a <Code>RelaxWorkChain node.
+        :param kwargs: any inputs that are specific to the plugin.
+        :return: a `aiida.engine.processes.ProcessBuilder` instance ready to be submitted.
+        """
+        protocol = protocol or self.get_default_protocol_name()
+
+        super().get_builder( # pylint: disable=too-many-function-args
+            structure,
+            engines,
+            protocol=protocol,
+            relax_type=relax_type,
+            electronic_type=electronic_type,
+            spin_type=spin_type,
+            magnetization_per_site=magnetization_per_site,
+            threshold_forces=threshold_forces,
+            threshold_stress=threshold_stress,
+            reference_workchain=reference_workchain,
+            **kwargs
+        )
+
+        if isinstance(electronic_type, str):
+            electronic_type = ElectronicType(electronic_type)
+        else:
+            electronic_type = ElectronicType(electronic_type.value)
+
+        if isinstance(relax_type, str):
+            relax_type = RelaxType(relax_type)
+        else:
+            relax_type = RelaxType(relax_type.value)
+
+        if isinstance(spin_type, str):
+            spin_type = SpinType(spin_type)
+        else:
+            spin_type = SpinType(spin_type.value)
+
+        # Get the protocol that we want to use
+        if protocol is None:
+            protocol = self._default_protocol
+        protocol = self.get_protocol(protocol)
+
+        # Set the builder
```

Comment:
Bit of a nitpick, but as a tip in general, comments like these are best left out. They don't really contribute anything as they literally just restate what the next line of code does. These code lines are really super self-explanatory so no comments is needed. Comments should really only be used when the code is not self-explanantory and explanation is needed. Often comments should therefore focus on the "why" and not the "what" or "how". In some cases the "how" is necessary since the code is not intuitive whatsoever, but that might actually be a warning that the code should be refactored and simplified.

Reply:
Very fair, these comments are not really needed - gotten rid of them in the latest commit.